libs/dh/metering-point/feature-search/src/lib/dh-can-activate-metering-point-overview.ts

```diff
@@ -28,12 +30,30 @@ export const dhCanActivateMeteringPointOverview: CanActivateFn = (
   route: ActivatedRouteSnapshot
 ) => {
   const meteringPointId: string = route.params[dhMeteringPointIdParam];
+  const isValidMP = dhIsValidMeteringPointId(meteringPointId);
+  const router = inject(Router);
 
-  return (
-    dhIsValidMeteringPointId(meteringPointId) ||
-    inject(Router).createUrlTree([
-      getPath<BasePaths>('metering-point'),
-      getPath<MeteringPointSubPaths>('search'),
-    ])
-  );
+  if (isValidMP) {
+    return query(DoesMeteringPointExistDocument, { variables: { meteringPointId } })
+      .result()
+      .then((result) => {
+        if (result.data === undefined) {
+          return router.createUrlTree(
```

Comment:
Issue: Looks like the function is returning either a string or a boolean?

Reply:
Have added a return type to the function.